### PR TITLE
Update deprecated config to typed config in istio-multicluster-ingressgateway EnvoyFilter template

### PR DESCRIPTION
**Pull request by kirecek**
This PR updates deprecated config spec in istio-multicluster-ingressgateway EnvoyFilter to use new typed_config.

Note: configs are no more present in proxy config dump as mentioned in https://github.com/istio/istio/issues/27204

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml [hunk 1]
```diff
@@ -46,7 +46,8 @@ spec:
       operation: INSERT_AFTER
       value:
         name: "envoy.filters.network.tcp_cluster_rewrite"
-        config:
+        typed_config:
```
Comment (howardjohn) [inline manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:49]: @nmittler what is the long term plan for this? We clearly don't have a test - are we planning to keep this around?

Comment (nmittler) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:49]: I'm actually not familiar with this ... is it needed for the `.global` configuration?

Comment (nmittler) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:49]: I guess the ingress needs this to make `.global` work? I'd be concerned about getting rid of this until folks are migrated over to the `cluster.local` way of doing things. We don't even have the new docs published yet.

Comment (kirecek) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:49]: > I guess the ingress needs this to make .global work? 

As far as I know, yes. Currently, we are using a setup like this:

```
+-----------------------------------------+
|MESH A                                   |
|                                         |
|    +----------------------+             |
|    |serviceentry          |             |
|    |                      |             |
|    |httpbin.demo.global   |             |
|    +----------+-----------+             |
|               |                         |
+-----------------------------------------+
                |
+-----------------------------------------+
|MESH B         |                         |
|               v                         |
|   +-----------+---------------------+   |
|   |ingress mtls gateway             |   |
|   +---------------------------------+   |
|   |envoyfilter                      |   |
|   |                                 |   |
|   |s/\.global$/.svc.cluster.local/  |   |
|   +-----------+---------------------+   |
|               |                         |
|               |                         |
|   +-----------v---------------------+   |
|   |service                          |   |
|   |                                 |   |
|   |httbin.demo                      |   |
|   +---------------------------------+   |
+-----------------------------------------+
```

Comment (GusTheSadGeek) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:49]: Without this fix I could not get Multicluster networking to work in 1.7.0+
All works correctly for me after I adjust the yaml as per the change detailed here.

**Review by nmittler — APPROVED** [review]

File: manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml (hunk 1 above)
Comment (tholcman) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:50]: Hi guys, any progress here? In 1.7+ Multicluster doesn't work according to the docs. Is it not a serious issue? 

@nmittler Can you maybe explain what is "the `cluster.local` way of doing things" you have mentioned ^? Or where can we read about that?

Comment (nmittler) [reply manifests/charts/gateways/istio-ingress/templates/preconfigured.yaml:50]: the new installation docs are here: https://preliminary.istio.io/latest/docs/setup/install/multicluster/

**Pull request merged**